ecs-cli/modules/config/readwriter_v1.go

```diff
@@ -66,48 +67,140 @@ func readINI(dest *Destination) (*CLIConfig, map[interface{}]interface{}, error)
+
+	// Try to read the config as YAML
+	clusterMap := make(map[interface{}]interface{})
+	errYAMLConfig := readYAML(configPath, clusterMap)
+	if errYAMLConfig == nil {
+		// cluster config exists and is YAML
+		processClusterMap(clusterConfig, clusterMap, configMap, cliConfig)
+	}
 
-	if errYAML := readYAML(path, configMap, to); errYAML == nil {
-		// File is YAML
-		return to, configMap, nil
-	} else if to, configMap, errINI := readINI(rdwr.destination); errINI == nil {
+	if _, err := os.Stat(profilePath); err != nil {
+		// credentials file exists- so that means we are using the new style configs
+		profileMap := make(map[interface{}]interface{})
+		if err = readYAML(profilePath, profileMap); err != nil {
+			return nil, nil, errors.Wrapf(err, "Error reading or parsing credentials file, %s", profilePath)
+		}
+
+		if err = processProfileMap(profileConfig, profileMap, configMap, cliConfig); err != nil {
+			return nil, nil, errors.Wrapf(err, "Error processing credentials file, %s", profilePath)
+		}
+		return cliConfig, configMap, nil
+	}
+
+	if to, configMap, errINI := readINI(rdwr.destination); errINI == nil {
 		// File is INI
 		return to, configMap, nil
-	} else if errINI != nil {
-		// Return the ini error
-		// This will return a parsing error for ini (format error)
-		return nil, nil, errINI
-	} else {
-		// If yaml through an error, but ini didn't throw an error- return the yaml error
-		return nil, nil, errYAML
 	}
+
+	if _, err := os.Stat(configPath); err != nil {
+		// Cluster config file exists, but it wasn't INI, so that means it was YAML but there was a parse error
+		return nil, nil, errYAMLConfig
+	}
+
+	// if no configs exist, we return empty objects
+	return cliConfig, configMap, nil
+
+}
+
+func processProfileMap(profileKey string, profileMap map[interface{}]interface{}, configMap map[interface{}]interface{}, cliConfig *CLIConfig) error {
+	if profileKey == "" {
+		var ok bool
+		if profileKey, ok = profileMap["default"].(string); !ok {
+			return errors.New("Format issue with profile config file; expected key not found.")
+		}
+	}
+	profile, ok := profileMap["ecs_profiles"].(map[interface{}]interface{})[profileKey].(map[interface{}]interface{})
+	if !ok {
+		return errors.New("Format issue with profile config file; expected key not found.")
+	}
+
+	configMap[awsAccessKey] = profile[awsAccessKey]
+	configMap[awsSecretKey] = profile[awsSecretKey]
+	if cliConfig.AWSAccessKey, ok = profile[awsAccessKey].(string); !ok {
+		return errors.New("Format issue with profile config file; expected key not found.")
+	}
+	if cliConfig.AWSSecretKey, ok = profile[awsSecretKey].(string); !ok {
+		return errors.New("Format issue with profile config file; expected key not found.")
+	}
+
+	return nil
+
+}
+
+func processClusterMap(clusterConfigKey string, clusterMap map[interface{}]interface{}, configMap map[interface{}]interface{}, cliConfig *CLIConfig) error {
+	if clusterConfigKey == "" {
+		var ok bool
```

Comment:
I was going to ask that in line 135 of this file, and it looks like it's bc the first value (`clusterConfigKey`) returned by checking the map is already passed in as an argument? which is why i asked if it was a good idea to mutate that argument. that is, instead of having:
```
var ok bool
fooArg, ok = theMap["key"].(string)
```
could you just have
```
fooCopy, ok := theMap["key"](string)
```
?